docs/android-core/pre-call/4-waiting-room.mdx

```diff
@@ -52,4 +76,27 @@ meeting.addSelfEventsListener(object : DyteSelfEventsListener {
 })
 ```
 
-Host can use [these methods to accept/reject participants](/android-core/participants#waiting-room-methods).
\ No newline at end of file
+  </TabItem>
+
+  <TabItem value="java" label="Java" default>
+
+```java
+meeting.addSelfEventsListener(new DyteSelfEventsListener() {
+    @Override
+    public void onWaitListStatusUpdate(waitListStatus: WaitListStatus) {
+        switch (waitListStatus) {
+            case WAITING:
+                // Local user is in the waiting room
+                break;
+            case REJECTED:
+                // Local user's join room request was rejected by the host
+                break;
+        }
+    }
+});
+```
+
+  </TabItem>
+</Tabs>
+
+Host can use [these methods to accept/reject participants](/android-core/participants#waiting-room-methods).
```

Comment:
(to-do): We will need to update this link after we update the Participants section.
MOB-1921